src/nvme/core.c

```diff
@@ -175,7 +175,7 @@ static int nvme_configure_sq(struct nvme_ctrl *ctrl, int qid, int qsize,
 		sq->dbbuf.eventidx = sqtdbl(ctrl->dbbuf.eventidxs, qid, dstrd);
 	}
 
-	len = pgmapn(&sq->pages.vaddr, qsize, __VFN_PAGESIZE);
+	len = pgmapn(&sq->pages.vaddr, qsize, 1ull << ctrl->config.mps);
```

Comment:
A comment on why it makes sense to use `mps` here would be nice.

Reply:
Comment added